proto/osmosis/concentrated-liquidity/pool-model/query.proto

```diff
@@ -154,7 +163,26 @@ message LiquidityDepthWithRange {
   ];
 }
 
-//=============================== TickLiquidityInBatches
+//=============================== LiquidityNetInDirection
+message QueryLiquidityNetInDirectionRequest {
+  uint64 pool_id = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
+  string token_in = 2 [ (gogoproto.moretags) = "yaml:\"token_in\"" ];
+  string bound_tick = 3 [
```

Comment:
How can we know we're requesting _all ticks_ (in a direction)? Should we just pass in the hardcoded next_tick value?

I'm thinking in the frontend, the estimates will go like:
* Try to reduce query size and _roughly_ estimate how many ticks we'll need based on size of trade and liquidity in pool.

Above is an edge case, as I want the above to work in most common cases (trading a moderate amount against a higher liquidity pool), this is an edge case scenario:
* If we run out of ticks in simulation iteration, we'll assume it's because we didn't request enough ticks. Then we'll try again with _all_ ticks in a given direction, by passing in either `maxTick` into `bound_tick` or using some special token/flag?? @mattverse @p0mvn 
* Only after we've tried getting the remaining tick net values, can we conclude the pool doesn't have enough liquidity to handle the user's current trade, where we will show an error.

Reply:
To estimate the expected final tick very roughly, we could take the current tick liquidity, take some proportion of that (e.g. 95%) because we assume that as we move away from the current tick, the liquidity decreases. Note, that we can come up with an algorithm to estimate the proportion too but can be hard coded to start. Finally, compute sqrt price delta in the following way:

- swapping y for x:

$$\Delta \sqrt P = \Delta y / L$$

- swapping x for y:

$$\Delta \sqrt P = L / \Delta x$$

Once we have that, we compute the "next expected sqrt price". From sqrt price, we can compute the expected final tick.

If that fails, we could then do some form of binary search where we take the estimated tick delta $$\Delta T$$  and if that wasn't enough, we query the next:
- $$2 \Delta T$$
- $$4 \Delta T$$

exclusively from the ticks we already received during prior querying to avoid redundant processing.

Similar to how Go slices reallocate buffer as we append to them.

Do you think we can keep the current design of querying until max bound for v1 and then work out a design for bounds estimates separately? @jonator @mattverse 

I think merging this would allow @mattverse and I to parallelize and further iterate on improvements

Let me know what you all think

Reply:
So we can run that calculation to get the next tick index then perhaps add a few extra ticks to avoid any rounding error?

Reply:
Yeah, that sounds right